### PR TITLE
openstack: remove default egress security rules

### PR DESCRIPTION
**Pull request by nhicher**
##### SUMMARY
With this change, it's possible for a user to remove
the default egress security rules during the
security group creation.

It's not possible to match these rules using
os_security_group_rule module without adding a specific
condition in _find_matching_rule function (with None
values for remote_ip_prefix, protocol and ethertype).

This solution looks easier to maintain.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
lib/ansible/modules/cloud/openstack/os_security_group.py

##### ANSIBLE VERSION
```
ansible 2.3.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Jun 26 2017, 10:20:05) [GCC 7.1.1 20170622 (Red Hat 7.1.1-3)]
```

### CODE REVIEW

**Comment by nhicher** [comment]
Any chance to have a review ?

**Comment by mnaser** [comment]
@nhicher do you mind rebasing this please?

**Comment by nhicher** [comment]
@mnaser done =)

File: lib/ansible/modules/cloud/openstack/os_security_group.py [hunk 1]
```diff
@@ -44,6 +44,14 @@
    availability_zone:
      description:
        - Ignored. Present for backwards compatibility
+     required: false
+   remove_default_rules:
```
Comment (mnaser) [inline lib/ansible/modules/cloud/openstack/os_security_group.py]:
```suggestion
   default_rules:
```

Comment (mnaser) [reply lib/ansible/modules/cloud/openstack/os_security_group.py]: Thinking more I think this would be easier to process

**Review by mnaser — CHANGES_REQUESTED** [review]

**Review by evrardjp — APPROVED** [review]
This looks okay, but I think we should move all those to use a cloud.network object instead.

**Comment by sshnaidm** [comment]
Thanks for submitting patch for Openstack Ansible modules!
We moved Openstack Ansible modules to Openstack repositories.
Next patches should be submitted not with Ansible Github but with
Openstack Gerrit: https://review.opendev.org/#/q/project:openstack/ansible-collections-openstack
Please submit your code there from now.
Thanks for your contribution and sorry for inconvienience.

**Comment by Akasurde** [comment]
Closing as per above.

**Pull request closed without merging**